server/mediaserver.go

```diff
@@ -1037,6 +1038,85 @@ func getPlaylistsFromStore(ctx context.Context, sess drivers.OSSession, manifest
 	return filesMap, jsonFiles, latestPlaylistTime, nil
 }
 
+func (s *LivepeerServer) streamMP4(w http.ResponseWriter, r *http.Request, jpl *core.JsonPlaylist, manifestID, track string) {
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	contentType, _ := common.TypeByExtension(".mp4")
+	w.Header().Set("Content-Type", contentType)
+	w.Header().Set("Connection", "keep-alive")
+
+	tc := ffmpeg.NewTranscoder()
+	or, ow, err := os.Pipe()
```

Comment:
Do we need to eventually Close() `or` too?

Reply:
Nice catch!
I've also changed how this function's exit handled - I think it was causing errors I've saw sometimes - when last couple of kb of resulting mp4 was not written to the network.
Now, before exit, it waits for Copy function to complete.